File: tests/fixtures/edgecase_repo/php/module4.php
<?php
namespace Example\Gen;

use Example\Gen\Base4;
use Example\Gen\Handler4 as H4;

class Base4
{
    protected $name;
}

interface Handler4
{
    public function handle($payload);
}

class Service4 extends Base4 implements Handler4
{
    public function handle($payload): bool
    {
        $this->validate($payload);
        return true;
    }

    private function validate($payload): void
    {
    }
}

function helper_4($value) {
    return $value;
}

File: tests/fixtures/edgecase_repo/dart/module4.dart
import 'dart:async';

import 'module3.dart';

abstract class Handler4 {
  bool handle(String payload);
}

class Base4 {
  final String name;
  Base4(this.name);
}

class Service4 extends Base4 implements Handler4 {
  Service4(String name) : super(name);

  @override
  bool handle(String payload) {
    this.validate(payload);
    return true;
  }

  void validate(String payload) {}
}

String helper4(String value) => value;
